csharp/src/Ice/TcpConnector.cs

```diff
@@ -6,9 +6,9 @@ namespace ZeroC.Ice
 {
     internal sealed class TcpConnector : IConnector
     {
-        public Connection Connect(string connectionId)
+        public Connection Connect(string connectionId, bool secure)
         {
-            ITransceiver transceiver = _endpoint.CreateTransceiver(this, _addr, _proxy);
+            ITransceiver transceiver = _endpoint.CreateTransceiver(this, _addr, _proxy, secure);
```

Comment:
I think we should keep preferNonSecure all the way here. Why this `!preferNonSecure`? 

If it's an ice2 endpoint, it should provide a secure transceiver when preferNonSecure = true and the peer's AcceptNonSecure is false.